react-app/src/components/Result.js

```diff
@@ -36,6 +36,48 @@ function Result(props) {
     return total;
   }
 
+  function getLabel() {
+    // attempt to get MRN and name with fhirpath
+    const mrn = fhirpath.evaluate(props.bundle, "Bundle.descendants().resource.where(resourceType='Patient').id")[0];
+    const name = fhirpath.evaluate(
+      props.bundle,
+      "Bundle.descendants().resource.where(resourceType='Patient').name.text",
+    )[0];
+
+    let isMasked = fhirpath.evaluate(
+      props.bundle,
+      "Bundle.descendants().resource.where(resourceType='Patient').identifier.extension.valueCode",
+    )[0];
```

Comment:
I think this could use a slight refactor to avoid 3 very similar fhirpath evaluates on the entire bundle. Notice the common chunk of the fhirpath expression that you're using to get the Patient resource. I'd suggest an approach where you use one evaluate to grab that patient resource. Then some of the other things should be trivial:

* Imagine you already have a patient resource when trying to access `.name.text`: do you even need fhirpath?
* The first argument to `fhirpath.evaluate` can be any json object, can you think of a way to do the `isMasked` logic with `fhirpath` but in a way that avoids repeating the chunk to find the patient? (assuming you already have a reference to the patient resource in the bundle at this point)

Reply:
I have refactored this code.